rclpy/rclpy/logging_service.py

```diff
@@ -12,17 +12,25 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
+
```

Comment:
Is this necessary?

Reply:
I believe so. Without it I get the following. 
def __init__(self, node: Node):
NameError: name 'Node' is not defined. Did you mean: 'None'?

I believe this case was discussed [`here`](https://peps.python.org/pep-0563/#runtime-annotation-resolution-and-type-checking) in PEP563.

Reply:
The thing is, we don't import that pretty much anywhere else.  So it seems unnecessary.  If we are getting that warning, then I think we probably need to just remove the `TYPE_CHECKING` block, no?  That is, unconditionally import `rclpy.Node`.

Reply:
If we unconditionally import rclpy.Node we get circular imports I believe.

ImportError: cannot import name 'Node' from partially initialized module 'rclpy.node' (most likely due to a circular import) (/opt/ros/iron/lib/python3.10/site-packages/rclpy/node.py)

Reply:
Actually, re-reading this I think we may be talking about different things.  Here I was referring to the `from __future__ import annotations`.  I don't think that is necessary.

Reply:
Ah yeah.  OK.  We've done the latter solution in other places, like in launch.  So yeah, let's go for that, thanks.

Reply:
> Actually, re-reading this I think we may be talking about different things. Here I was referring to the `from __future__ import annotations`. I don't think that is necessary.

Without the `__future__ import annotations` at run time the constructor fails because Node is undefined. The `__future__ import annotations` converts all type annotations to their string version eg `node: Node` becomes `node: 'Node'`